tensordict/tensordict.py

```diff
@@ -1162,15 +1199,19 @@ def unbind(self, dim: int) -> Tuple[TensorDictBase, ...]:
         if dim < 0:
             dim = self.batch_dims + dim
         batch_size = torch.Size([s for i, s in enumerate(self.batch_size) if i != dim])
-        return tuple(
-            self.apply(
-                lambda tensor, idx=_idx: tensor[idx],
-                batch_size=batch_size,
-                _is_shared=self.is_shared(),
-                _is_memmap=self.is_memmap(),
+        out = []
+        for _idx in idx:
+            out.append(
+                self.apply(
+                    lambda tensor, idx=_idx: tensor[idx],
+                    batch_size=batch_size,
+                )
             )
-            for _idx in idx
-        )
+            if self.is_shared():
+                out[-1].share_memory_()
+            elif self.is_memmap():
+                out[-1].memmap_()
```

Comment:
@tcbegley 
Here we index the memmap tensor, hence we get a memmap tensor.
But the tensordict is not marked as memmap.
so we call memmap_() if needed, to "mark" it as memmap. But we don't expect anything to happen to the tensors (they should stay where they are). What would happen with #189?

Reply:
The way I have things set up at the moment, if the entries of the tensordict are already `MemmapTensors`, _and_ either the user doesn't specify a `prefix` in `memmap_`, or the memmaps are already located in the right place, then `memmap_` is a no-op on the entries, it simply sets `self._is_memmap = True` and locks the tensordict.

So if I'm understanding you right, there shouldn't be any conflict here I think.